### PR TITLE
Fix sample j fn

### PR DESCRIPTION
**Pull request by amo-susi**
Fixed zero division error again.

### CODE REVIEW

**Review by slyrus — COMMENTED** [review]
It's been a while since I've looked at this. Any chance you've got a test or at least a description of what this is doing?

File: imageops.lisp [hunk 1]
```diff
@@ -191,7 +191,8 @@
   (let ((len (reduce #'* (array-dimensions array))))
     (if force-simple
         (let* ((disp (make-array len
-                                 :displaced-to array))
+                                 :displaced-to array
+				 :element-type (array-element-type array)))
```
Comment (phoe) [inline imageops.lisp:195]: @slyrus This looks like an optimization - the created array will not be a T array but will be specialized to the AET of the original array.

File: imageops.lisp [hunk 2]
```diff
@@ -200,7 +201,7 @@
              do (setf (row-major-aref dest i)
                       (elt vec i)))
           dest)
-        (let* ((disp (make-array len :displaced-to array)))
+        (let* ((disp (make-array len :displaced-to array :element-type (array-element-type array))))
```
Comment (phoe) [inline imageops.lisp:204]: @slyrus Ditto.

File: thresholding.lisp [hunk 3]
```diff
@@ -68,7 +68,9 @@ The elements at indexex 0 and 255 are set to +j-fn-max-value+."
            (incf apriori-1 (aref histogram idx))
            (if (zerop apriori-1)
                (setf (aref samples idx) +j-fn-max-value+)
-               (let* ((apriori-2 (- 1d0 apriori-1))
+               (let* ((apriori-2 (if (= apriori-1 1.0d0)
+				     least-positive-double-float
+				     (- 1d0 apriori-1)))
```
Comment (phoe) [inline thresholding.lisp:73]: @slyrus If `apriori-1` is equal to 1d0, then `apriori-2` is going to be equal to 0d0. This means that computing `mean-2` like this:

```lisp
                      (mean-2 (/ (loop for i from (1+ idx) to (- histlen 1)
                                    summing (* (aref histogram i) i))
                                 apriori-2))
```

is going to divide by zero. This patch avoids it by instead nudging that zero upwards into a machine epsilon.

**Review by phoe — APPROVED** [review]

**Pull request merged**